src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,625 @@
+                case BY_LEXICOGRAPHICAL_DESCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return String.CASE_INSENSITIVE_ORDER.compare(o2.getName(), o1.getName());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        result.add(file);
+                    }
+                    return result;
+
+
+                case BY_LAST_UPDATE_DESCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return Long.valueOf(o1.lastModified()).compareTo(o2.lastModified());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        result.add(file);
+                    }
+                    return result;
+                case BY_LAST_UPDATE_ASCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return Long.valueOf(o2.lastModified()).compareTo(o1.lastModified());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        result.add(file);
+                    }
+                    return result;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String fileInfo(File file) {
+        String divider = "|";
+        long size;
+        String str2;
+
+        if (file.isDirectory()) {
+            size = 0;
+        } else {
+            size = file.length();
+        }
+
+        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
+        String str1 = sdf.format(file.lastModified());
+        if (file.isDirectory()) str2 = file.getName() + "/";
+        else str2 = file.getName();
+
+        return String.format("%-20s %15s %s %n", str1 + divider, size + divider, str2);
+    }
+
+    /**
+     * Create new file within current directory with given name
+     *
+     * @param name name of new file
+     * @throws FileAlreadyExistException if file or directory with given name already exists
+     * @throws FileManagerException      if any other problem happened
+     */
+    @Override
+    public void createFile(String name) throws FileAlreadyExistException, FileManagerException {
```

Comment:
This method **MUST** throw exception and do not print some messages about exceptions.